tests/integration/utils.go

```diff
@@ -723,55 +731,53 @@ func setStartingNodes(
 	}
 }
 
-func restartNodes(
-	s *state,
-) {
-	if s.dbt == badgerIMType || s.dbt == defraIMType {
-		return
-	}
-	closeNodes(s)
-
+func startNodes(s *state, action Start) {
+	nodes := getNodes(action.NodeID, s.nodes)
 	// We need to restart the nodes in reverse order, to avoid dial backoff issues.
-	for i := len(s.nodes) - 1; i >= 0; i-- {
+	for i := len(nodes) - 1; i >= 0; i-- {
+		nodeIndex := i
+		if action.NodeID.HasValue() {
+			nodeIndex = action.NodeID.Value()
+		}
```

Comment:
info: When I fix the `getNodes` here https://github.com/sourcenetwork/defradb/issues/3076 won't have to do this hack anymore, I am guessing this is because the nodeIndex is wrong with their is a nodeID specified correct?

Reply:
yes that's pretty much it.